spotify_project/spotify/base.py

```diff
@@ -39,65 +39,11 @@ def fetch_current_user(self):
         return spotify_user
 
     def pull_library_data(self, user):
-        tracks = []
-        for track in self.tracks:
-            tracks.append(self.add_track_to_db(track))
-        self.add_tracks_to_library(user, tracks)
+        track_dicts = list(self.tracks)
+        ArtistUtils().add_artists_to_db(track_dicts)
```

Comment:
It would be worth building instances of each of these util classes in `__init__`. Then using them here. Something like:

```python
self.artist_utils = ArtistUtils()
...
self.artist_utils.add_artists_to_db(...)
```

This is useful for a few reasons. It's a touch cleaner to read, we aren't rebuilding the object for each method we call on it. If the classes were doing some kind of caching, by rebuilding each time you use it, it would not actually use the cached values. Not applicable here, but still something to consider.